android-sdk/src/main/java/com/checkout/android_sdk/Utils/CheckoutTheme.kt

```diff
@@ -0,0 +1,69 @@
+package com.checkout.android_sdk.Utils
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.Resources
+import android.util.TypedValue
+import androidx.appcompat.widget.ThemeUtils.getThemeAttrColor
+import com.checkout.android_sdk.R
+
+internal class CheckoutTheme(val context: Context) : CheckoutThemeInterface {
+    private var colorPrimary: MutableMap<String, Any?>
+    private var colorAccent: MutableMap<String, Any?>
+    private var colorButtonNormal: MutableMap<String, Any?>
+    private var colorControlNormal: MutableMap<String, Any?>
+    private var textColorPrimary: MutableMap<String, Any?>
+    private var colorControlActivated: MutableMap<String, Any?>
+
+    init {
+        colorPrimary = geThemeColorResource(R.attr.colorPrimary)
+        colorAccent = geThemeColorResource(R.attr.colorAccent)
+        colorButtonNormal = geThemeColorResource(R.attr.colorButtonNormal)
+        colorControlNormal = geThemeColorResource(R.attr.colorControlNormal)
+        textColorPrimary = geThemeColorResource(android.R.attr.textColorPrimary)
+        colorControlActivated = geThemeColorResource(R.attr.colorControlActivated)
+    }
+
+    @SuppressLint("RestrictedApi")
```

Comment:
Question : Any reason to suppress the lint here ?

Reply:
I was using getThemeColorAttr method (utility method for getting theme colours from attributes)of the androidx library, which may be subject to change in future releases of androidx versions. The reason for using supresslint is to avoid conflict of API change in future for this specific method for the safe call. However, I have removed that and applied the core native method to get the colours. I will keep a note to not use as much as public API methods which aim to change in the future.